src/System.Net.Http/tests/UnitTests/HttpProxy.cs

```diff
@@ -0,0 +1,157 @@
+        {
+            RemoteInvoke(() =>
+            {
+
+                IWebProxy p;
+                Uri u;
+
+                // It should not return object if there are no variables set.
+                Assert.True(HttpEnvironmentProxy.TryToCreate() == null);
+
+                Environment.SetEnvironmentVariable("all_proxy", "http://1.1.1.1:3000");
+                p = HttpEnvironmentProxy.TryToCreate();
+                Assert.True(p != null);
+                Assert.True(p.Credentials == null);
+
+                u = p.GetProxy(fooHttp);
+                Assert.True(u != null && u.Host == "1.1.1.1");
+                u = p.GetProxy(fooHttps);
+                Assert.True(u != null && u.Host == "1.1.1.1");
+
+                Environment.SetEnvironmentVariable("http_proxy", "http://1.1.1.2:3001");
+                p = HttpEnvironmentProxy.TryToCreate();
+                Assert.True(p != null);
+
+                // Protocol specific variables should take precedence over all_
+                // and https should still use all_proxy.
+                u = p.GetProxy(fooHttp);
+                Assert.True(u != null && u.Host == "1.1.1.2" && u.Port == 3001);
+                u = p.GetProxy(fooHttps);
+                Assert.True(u != null && u.Host == "1.1.1.1" && u.Port == 3000);
+
+                // Set https to invalid strings and use only IP & port for http.
+                Environment.SetEnvironmentVariable("http_proxy", "1.1.1.3:3003");
+                Environment.SetEnvironmentVariable("https_proxy", "ab!cd");
+                p = HttpEnvironmentProxy.TryToCreate();
+                Assert.True(p != null);
+
+                u = p.GetProxy(fooHttp);
+                Assert.True(u != null && u.Host == "1.1.1.3" && u.Port == 3003);
+                u = p.GetProxy(fooHttps);
+                Assert.True(u != null && u.Host == "1.1.1.1" && u.Port == 3000);
+
+                // Try valid URI with unsupported protocol. It will be ignored
+                // to mimic curl behavior.
+                Environment.SetEnvironmentVariable("https_proxy", "socks5://1.1.1.4:3004");
+                p = HttpEnvironmentProxy.TryToCreate();
+                Assert.True(p != null);
+                u = p.GetProxy(fooHttps);
+                Assert.True(u != null && u.Host == "1.1.1.1" && u.Port == 3000);
+
+                // Set https to valid URI but different from http.
+                Environment.SetEnvironmentVariable("https_proxy", "http://1.1.1.5:3005");
+                p = HttpEnvironmentProxy.TryToCreate();
+                Assert.True(p != null);
+
+                u = p.GetProxy(fooHttp);
+                Assert.True(u != null && u.Host == "1.1.1.3" && u.Port == 3003);
+                u = p.GetProxy(fooHttps);
+                Assert.True(u != null && u.Host == "1.1.1.5" && u.Port == 3005);
+
+                return SuccessExitCode;
+            }).Dispose();
+        }
+
+        [Fact]
+        public void parseCredientials()
```

Comment:
spelling and case:  ParseCredentials().

Also, you should follow the 3-part naming convention for test methods.  See the rest of the Http tests.